### PR TITLE
refactor: E2e Smoke Tests to Execute Across Multiple Virtual Machines on Bitrise

### PR DESCRIPTION
**Pull request by Andepande**
## **Description**

This pull request brings improvements to the smoke test execution process on Bitrise, leading to parallel test runs across multiple virtual machines. These enhancements aim to:

- **Reduced Run Duration:** Cut down the time required for e2e runs on Bitrise by executing smoke tests concurrently across various virtual machines.
- **Simplified Fault Identification:** By segregating different areas, it becomes simpler to identify test failures within their respective functional domains.
- **Workflow Refinement:** By isolating distinct areas, the necessity to retry workflows decreases, concentrating solely on specific areas where tests failed.

### Changes Introduced

- Implemented parallel test runs across multiple virtual machines to optimise the smoke test execution process on Bitrise.
- Introduced new smoke tags (**smokeConfirmations**, **smokeSwaps**, **smokeCore**) to categorise and streamline testing efforts targeting specific functional areas.

These enhancements will significantly contribute to improving test efficiency, ensuring a more comprehensive evaluation of distinct functionalities via targeted smoke tests.

### Observations for Discussion::

**Bitrise Pipeline with Three Workers on Detox**
   - Credits Usage: ~622 credits
   - Run Time: ~55 minutes
   - [Pipeline Link](https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/444ec885-1f88-4870-b762-9c9b29961ae0)

2. **Pipeline Utilising Two Workers**
   - Credits Usage: ~556 credits
   - Run Time: ~45 minutes
   - [Pipeline Link](https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/4668d6db-fb8d-4833-933f-13a2a0fe9ad1)

If we reduce the number of parallel workers used to run tests in detox, builds may take longer and utilise more credits, even with optimised utilisation of virtual machines.

- 

## **Related issues**

Fixes: https://app.zenhub.com/workspaces/metamask-mobile-5f984938ddc0e4001d4b79cb/issues/gh/metamask/metamask-mobile/7844

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -597,6 +649,10 @@ workflows:
                 else
                   TEST_SUITE="Smoke"
                 fi
+                if [ -n "$TEST_SUITE_TAG" ]; then
+                echo "TEST_SUITE_TAG value is: $TEST_SUITE_TAG"
```
Comment (cortisiko) [inline bitrise.yml:653]: You have this in for debugging purposes? If not probably remove

File: bitrise.yml [hunk 2]
```diff
@@ -597,6 +649,10 @@ workflows:
                 else
                   TEST_SUITE="Smoke"
```
Comment (cortisiko) [inline bitrise.yml:650]: we no longer use this flag anymore, perhaps remove it

**Review by cortisiko — CHANGES_REQUESTED** [review]
small comments

File: bitrise.yml (hunk 1 above)
Comment (Andepande) [reply bitrise.yml:653]: I'll keep in for now as I'm using it to pass the tag type from the bitrise env variable, once I work on condensing the regression I'll refine it some more

**Review by sethkfman — DISMISSED** [review]
LGTM

**Review by cortisiko — APPROVED** [review]

**Pull request merged**